### DISCUSSION TITLE
Whatever happened to dotnet build integration? docfx.console?

### DISCUSSION THREAD

**Discussion by craigfowler** [dotnet/docfx#8415] [Q&A] [answered]
I've got a few old-ish open source .NET projects which generated their documentation as part of the build process via the `docfx.console` NuGet package. The DocFX dotnet tool _was not installed_ on the CI environment but the docs were being generated just fine during the build. I had just installed the NuGet package & added DocFX config; when I used `dotnet build` on the CI environment I'd receive docs which I could copy to my GitHub Pages site and auto-commit.

I took some time off open source development (around 6-9 months) but I've come back to them and oddly this mechanism is no longer working. I had presumed that this might be because the CI environments had upgraded to .NET 7 and yet I'd left the docfx.console package at a lower version (2.58.9 fwiw). I added a `global.json` to pin the SDK to .NET 6 but that hasn't fixed it.

What's more, I went to the current DocFX documentation and there's no mention of building the docs this way anymore (`dotnet build` integration). Is this no longer supported/withdrawn? Have I come from the multiverse of madness and this was never supposed to work? I'm really confused as to why a build feature I've been using for a while has mysteriously stopped working 🤔

Any insights?

**Comment by KalleOlaviNiemitalo** [comment, marked as answer]
The `docfx.console` package is no longer produced. There are instead:

- The [docfx](https://www.nuget.org/packages/docfx) .NET tool package. It only provides the tool and does not automatically integrate with your project build to run the tool. You could add custom MSBuild targets to your .NET projects to run the `docfx` tool, or you could make your continuous-integration build script run first MSBuild and then separately the `docfx` tool.
- The [Microsoft.DocAsCode.App](https://www.nuget.org/packages/Microsoft.DocAsCode.App) library package. You can set up an executable .NET project that references this and calls [Docset.Build](https://github.com/dotnet/docfx/blob/ed77a7c7454fe73bf987ec672be88c270ceaeb96/src/Microsoft.DocAsCode.App/Docset.cs#L34). But then you'd again need something to automate running the resulting executable.

[Quick Start](https://dotnet.github.io/docfx/index.html) suggests installing the latest `docfx` as a global tool, but I prefer installing a known version as a local tool instead.

**Reply by craigfowler** [reply]
Thanks for that.  I had wondered if this might be the case.

I wish that there had been better signposting of this on the docs site 😞 Rather than just removing mention of it, it would have been useful to say _"Hey, you used be able to do X but now you can't"_.  Certainly would have saved me some time and astonishment.

**Comment by lanthonyneville** [comment]
After struggling with this myself, I wrote up how I did this here: #8605
